File: backend/src/main/java/com/back/ourlog/external/spotify/dto/SpotifySearchResponse.java
package com.back.ourlog.external.spotify.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Getter
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true) // 전체 응답에서 불필요한 필드 무시
public class SpotifySearchResponse {
    private Tracks tracks;

    @Getter
    @NoArgsConstructor
    public static class Tracks {
        private List<TrackItem> items;
    }
}

File: backend/src/main/java/com/back/ourlog/external/spotify/dto/SpotifySearchResponse.kt
package com.back.ourlog.external.spotify.dto

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@JsonIgnoreProperties(ignoreUnknown = true)
data class SpotifySearchResponse(
    val tracks: Tracks? = null
) {
    @JsonIgnoreProperties(ignoreUnknown = true)
    data class Tracks(
        val items: List<TrackItem?>? = null
    )
}
